### PR TITLE
Error if bumping past max_entry_expiration and and host function to retrieve max_entry_expiration

### LINKED ISSUE #973: Enforce max expiration ledger
Currently we allow bumping entries by arbitrary amounts, but silently clamp the values at max expiration ledger. This will lead to footguns in case of temporary entries that have to live for a certain time period (such as nonces for expirable signatures). To solve this, we need to do 2 things:

- Provide max expiration ledger (or max bump amount) as a host function, so that contracts could make decisions (like we do to reject signatures/token approvals that expire too far into future)
- Return error for bumps that try bumping entry past the max expiration ledger (both for bump op and bump host fn)

### PR DESCRIPTION
**Pull request by sisuresh**
### What

Resolves #973.

### CODE REVIEW

File: soroban-env-common/env.json [hunk 1]
```diff
@@ -120,6 +120,13 @@
                     "args": [],
                     "return": "AddressObject",
                     "docs": "Get the Address object for the current contract."
+                },
+                {
+                    "export": "a",
+                    "name": "get_max_entry_expiration",
+                    "args": [],
+                    "return": "U32Val",
+                    "docs": "Return the maximum number of ledgers from the current ledger that an entry can live for."
```
Comment (dmkozh) [inline soroban-env-common/env.json:129]: We probably should also mention that the equation is `current ledger + max ledger - 1`.

Comment (sisuresh) [reply soroban-env-common/env.json]: Updated to expose `max_expiration_ledger` instead.

File: soroban-env-host/src/storage.rs [hunk 2]
```diff
@@ -335,9 +335,19 @@ impl Storage {
...
+
+        let new_expiration =
+            host.with_ledger_info(|li| Ok(li.sequence_number.saturating_add(bump_by_ledgers)))?;
+
+        if new_expiration > host.max_expiration_ledger()? {
+            return Err(host.err(
+                ScErrorType::Storage,
+                ScErrorCode::InvalidAction,
+                "trying to bump past max expiration ledger",
+                &[],
```
Comment (dmkozh) [inline soroban-env-host/src/storage.rs:347]: Add `new_expiration` to error args?

**Review by dmkozh — APPROVED** [review]

File: soroban-env-common/env.json [hunk 3]
```diff
@@ -120,6 +120,13 @@
                     "args": [],
                     "return": "AddressObject",
                     "docs": "Get the Address object for the current contract."
+                },
+                {
+                    "export": "a",
+                    "name": "get_max_expiration_ledger",
+                    "args": [],
+                    "return": "U32Val",
+                    "docs": "Returns the max ledger sequence that an entry can live to."
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply soroban-env-common/env.json]: Done

**Pull request merged**